fecgov/fec-cms
# Feature/2744 aggregate totals box

## Summary
Adding the aggregate totals box!

- Resolves #2744 

Removed the macro for aggregate-totals-box from an earlier commit.
Simplified how the module is included in data/templates/macros/bythenumbers.jinja.

## Impacted areas of the application
- The new box can be seen on the raising and spending bythenumbers pages.
- It has its own css—its scss was added to static/scss/data-landing.scss
- Its JavaScript is isolated but has a new entry in webpack.config.js

## Screenshots
![image](https://user-images.githubusercontent.com/26720877/55028773-f5ff6400-4fde-11e9-88a1-2e3cfccc5e3a.png)
![image](https://user-images.githubusercontent.com/26720877/55028821-0ca5bb00-4fdf-11e9-9064-c7dcb374395c.png)
![image](https://user-images.githubusercontent.com/26720877/55028842-1af3d700-4fdf-11e9-91c6-16523bcea524.png)

## Related PRs
branch | PR
------ | ------ 
feature/2562/embed-bar-charts-election-map-homepage | #2748 

## How to test
- Set the feature flag environment variable `FEC_FEATURE_AGGR_TOTS=1`
- Go to either [raising](http://localhost:8000/data/raising-bythenumbers/) or [spending](http://localhost:8000/data/spending-bythenumbers) bythenumbers.
- The aggregate totals box should only appear after the rest of the page has loaded and only if there's data to display (give it about a second).
- When the animation is complete, check the console log to make sure the query response results total_disbursements (or total_receipts) match the dollar figure on the screen at the end of the animation.
- Change a pull-down, let the animation tick, compare the value with the console values.
- Change the other pull-down, let the animation play, compare the value again.
- Make sure the description text updates for each pull-down change.

Notes:
- The number of animation steps/ticks is randomized each time the page is loaded but they should remain the same through each pull-down change.
- The value increment for the first animation (effectively building from $0) should be <$100 per animation step/timer tick.
- The value increment for subsequent animations should be whatever it takes to get from the previous value to the new value in this session's number of steps (e.g., if there are three steps, each one should be a third of the difference between the previous and next values.). These steps could be negative when moving from a larger to a smaller value.
____
